src/main/java/com/amazonaws/encryptionsdk/keyrings/StandardKeyrings.java

```diff
@@ -31,48 +28,74 @@ private StandardKeyrings() {
-     * @param keyName           A value that, together with the key namespace, identifies the wrapping key.
-     * @param publicKey         The RSA public key used by this keyring to encrypt data keys.
-     * @param privateKey        The RSA private key used by this keyring to decrypt data keys.
-     * @param wrappingAlgorithm The RSA algorithm to use with this keyring.
-     * @return The {@link Keyring}
+     * @return The {@link RawRsaKeyringBuilder}
      */
-    public static Keyring rawRsa(String keyNamespace, String keyName, PublicKey publicKey, PrivateKey privateKey, String wrappingAlgorithm) {
-        return new RawRsaKeyring(keyNamespace, keyName, publicKey, privateKey, wrappingAlgorithm);
+    public static RawRsaKeyringBuilder rawRsa() {
+        return RawRsaKeyringBuilder.standard();
     }
       
     /**  
      * Constructs a {@code Keyring} which interacts with AWS Key Management Service (KMS) to create,
-     * encrypt, and decrypt data keys using KMS defined Customer Master Keys (CMKs).
+     * encrypt, and decrypt data keys using the supplied AWS KMS defined Customer Master Key (CMK).
+     * Use {@link #awsKms()} for more advanced configuration using a {@link AwsKmsKeyringBuilder}/
      *
-     * @param clientSupplier    A function that returns a KMS client that can make GenerateDataKey,
-     *                          Encrypt, and Decrypt calls in a particular AWS region.
-     * @param grantTokens       A list of string grant tokens to be included in all KMS calls.
-     * @param keyIds            A list of strings identifying KMS CMKs, in ARN, CMK Alias, or ARN Alias format.
-     * @param generator         A string that identifies a KMS CMK responsible for generating a data key,
-     *                          as well as encrypting and decrypting data keys in ARN, CMK Alias, or ARN Alias format.
+     * @param generatorKeyId    An {@link AwsKmsCmkId} in ARN, CMK Alias, ARN Alias or Key Id format that identifies a
+     *                          AWS KMS CMK responsible for generating a data key, as well as encrypting and
+     *                          decrypting data keys .
      * @return The {@code Keyring}
      */
-    public static Keyring kms(KmsClientSupplier clientSupplier, List<String> grantTokens, List<String> keyIds, String generator) {
-        return new KmsKeyring(DataKeyEncryptionDao.kms(clientSupplier, grantTokens), keyIds, generator);
+    public static Keyring awsKms(AwsKmsCmkId generatorKeyId) {
+        return AwsKmsKeyringBuilder.standard()
+                .generatorKeyId(generatorKeyId)
+                .build();
+    }
+
+    /**
+     * Returns a {@link AwsKmsKeyringBuilder} for use in constructing a keyring which interacts with
+     * AWS Key Management Service (KMS) to create, encrypt, and decrypt data keys using AWS KMS defined
+     * Customer Master Keys (CMKs).
+     *
+     * @return The {@link AwsKmsKeyringBuilder}
+     */
+    public static AwsKmsKeyringBuilder awsKms() {
```

Comment:
Nit: The lack of symmetry in these three methods is a little odd, given one produces the keyring itself and the other two builders. Perhaps add "...Builder" to the latter two? And perhaps to the two raw methods above too in that case?

Reply:
I've renamed the builder methods to ...Builder